ts/services/ElementService.ts

```diff
@@ -0,0 +1,185 @@
+import * as Bluebird from 'bluebird';
+import * as fs from 'fs';
+import * as request from 'request-promise';
+
+const MAINT_TAG_NAME = 'n.state.maintenance';
+const MAINT_END_TAG_NAME = 'n.state.maintenance_end';
+const MS_IN_HOUR = 3600000;
+
+class ElementService {
+
+  public async lsMaintenanceMode(config, logger): Promise<void> {
+    logger.info('\nListing elements in maintenance mode');
+    try {
+      const response = await request({
```

Comment:
Re-adding so it doesn't get lost:

Reply:
I'm going to be refactoring a bit in the next PR, I'll handle this observation there.